frontend/packages/dev-console/src/components/helm/HelmReleaseUpgradePage.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import * as _ from 'lodash';
+import { safeLoad } from 'js-yaml';
+import { coFetchJSON, coFetch } from '@console/internal/co-fetch';
+import { LoadingBox } from '@console/internal/components/utils';
+import { HelmRelease, HelmChartMetaData, HelmActionType } from './helm-types';
+import { getChartURL } from './helm-utils';
+import HelmActionPage from './HelmActionPage';
+
+type HelmReleaseUpgradePageProps = RouteComponentProps<{ ns: string; releaseName: string }>;
+
+export const HelmReleaseUpgradePage: React.FC<HelmReleaseUpgradePageProps> = (props) => {
+  const {
+    match: {
+      params: { ns: namespace, releaseName },
+    },
+  } = props;
+
+  const [chartValues, setChartValues] = React.useState({});
+  const [chartName, setChartName] = React.useState('');
+  const [helmChartData, setHelmChartData] = React.useState<HelmChartMetaData[]>([]);
+  const [chartValuesLoaded, setChartValuesLoaded] = React.useState(false);
+  const [chartVersions, setChartVersions] = React.useState({});
+  const [activeChartVersion, setActiveChartVersion] = React.useState<string>('');
+
+  React.useEffect(() => {
+    const getChartDetails = () => {
+      coFetchJSON(`/api/helm/release?ns=${namespace}&release_name=${releaseName}`)
+        .then((res) => {
+          setActiveChartVersion(res.chart.metadata.version);
```

Comment:
Possible data leak when the component is unmounted. Follow the older implementation.